add kexp and kget-exp as replacement for removed kubectl `--export` flag

toschneck opened the pull request:
For creating a fast template of current deployment manifests without metadata and status fields:
```
#fuzzy option
kget-exp pod > pod.yaml

#dedicated usage
k get pod NAME | kexp | pod.yaml
```

toschneck committed:
add kget-ex as replacement

for 'kubectl get -o yaml --export' flag combi

toschneck commented:
/assign @kron4eg

toschneck committed:
kget-ex remove also  kubectl.kubernetes.io/last-applied-configuration values

toschneck committed:
add cosmetic whitespaces

kron4eg commented on fubectl.source:
```diff
@@ -86,6 +86,18 @@ kget() {
 
+# [kget-ex] get a resource by its YAML as former `--export` flag
+kget-ex() {
+    kget "$@" \
+    | yq d - 'metadata.resourceVersion' \
+    | yq d - 'metadata.uid' \
+    | yq d - 'metadata.creationTimestamp' \
+    | yq d - 'metadata.selfLink' \
+    | yq d - 'metadata.managedFields' \
+    | yq d - 'metadata.annotations.kubectl*' \
+    | yq d - 'status'
```
```suggestion
    kget "$@" |
        yq d - 'metadata.resourceVersion' |
        yq d - 'metadata.uid' |
        yq d - 'metadata.creationTimestamp' |
        yq d - 'metadata.selfLink' |
        yq d - 'metadata.managedFields' |
        yq d - 'metadata.annotations.kubectl*' |
        yq d - 'status'
```

toschneck committed:
Update fubectl.source

Co-authored-by: Artiom Diomin <artiom@hey.com>

toschneck commented:
I have on further improvement, please wait before merge

toschneck committed:
improve usage
- add separate `kexp` function for dedicated usage like `k get pod xyz | kexp`
- rename export function to `kget-exp`

toschneck commented:
@kron4eg thanks for the improvement, please review again :smiley:

kron4eg commented on fubectl.source:
```diff
@@ -294,3 +315,17 @@ khelp() {
     fi
 }
+
+#### CUSTOM TOBI
```
@toschneck is this an accident or you are willing to include this too?

toschneck replied:
@kron4eg fixed it

toschneck committed:
remove unwanted commits

kron4eg commented:
Few days again I've stumbled upon this:

```
$ kubectl krew info neat
NAME: neat
INDEX: default
URI: https://github.com/itaysk/kubectl-neat/releases/download/v2.0.1/kubectl-neat_darwin.tar.gz
SHA256: c1e8ac967232173f9a2cadfb40af342363f9d2cf9e8bad02e025da6300468f84
VERSION: v2.0.1
HOMEPAGE: https://github.com/itaysk/kubectl-neat
DESCRIPTION:
If you try to `kubectl get` resources you have just created,
they be unreadably verbose. `kubectl-neat` cleans that up by
removing default values, runtime information, and other internal fields.
Examples:
`$ kubectl get pod mypod -o yaml | kubectl neat`
`$ kubectl neat get -- pod mypod -o yaml`
```

toschneck committed:
Merge remote-tracking branch 'origin/master' into kget-ex

toschneck committed:
kexp used - kubectl neat instead yq

toschneck commented:
@kron4eg  replaced it with `kubectl neat` seems to work good :-)
